kernel/aster-nix/src/thread/exception.rs

```diff
@@ -5,67 +5,67 @@
 use ostd::{cpu::*, mm::VmSpace};
 
 use crate::{
-    prelude::*, process::signal::signals::fault::FaultSignal,
-    vm::page_fault_handler::PageFaultHandler,
+    prelude::*,
+    process::signal::signals::fault::FaultSignal,
+    vm::{page_fault_handler::PageFaultHandler, perms::VmPerms},
 };
 
+/// This trait is implemented by [`CpuExceptionInfo`] which can tell the information of a page
+/// fault.
+pub trait PageFaultContext {
+    /// Whether it's a page fault
+    fn is_page_fault(&self) -> bool;
+
+    /// The virtual address where a page fault occurred.
+    fn page_fault_address(&self) -> Vaddr;
+
+    /// The VmPerms required by the memory operation that causes page fault.
+    /// For example, a "store" operation may require `VmPerms::WRITE`.
+    fn page_fault_required_perms(&self) -> VmPerms;
+}
+
 /// We can't handle most exceptions, just send self a fault signal before return to user space.
 pub fn handle_exception(ctx: &Context, context: &UserContext) {
     let trap_info = context.trap_information();
-    let exception = CpuException::to_cpu_exception(trap_info.id as u16).unwrap();
-    log_trap_info(exception, trap_info);
-    let root_vmar = ctx.process.root_vmar();
+    log_trap_info(trap_info);
 
-    match *exception {
-        PAGE_FAULT => {
-            if handle_page_fault(root_vmar.vm_space(), trap_info).is_err() {
-                generate_fault_signal(trap_info);
-            }
-        }
-        _ => {
-            // We current do nothing about other exceptions
-            generate_fault_signal(trap_info);
+    if trap_info.is_page_fault() {
+        let current = current!();
```

Comment:
Use `current = ctx.process` would be faster here. The `current!` macro is a bit slow. This optimization is introduced in #1126 .